doc/developers/minimal_reproducer.rst

```diff
@@ -0,0 +1,404 @@
+            "target": [72, 55, 32, 43],
+        }
+    )
+    X = df[["feature_name"]]
+    y = df["target"]
+
+    gbdt = GradientBoostingRegressor()
+    gbdt.fit(X, y) # compiles
+    gbdt = GradientBoostingRegressor(n_iter_no_change=5)
+    gbdt.fit(X, y) # raises warning
+
+
+Use markdown
+------------
+
+To format code or text into its own distinct block, use triple backticks.
+`Markdown
+<https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax>`_
+supports an optional language identifier to enable syntax highlighting in your
+fenced code block. For example::
+
+    ```python
+    from sklearn.datasets import make_blobs
+
+    n_samples = 100
+    n_components = 3
+    X, y = make_blobs(n_samples=n_samples, centers=n_components)
+    ```
+
+will render a python formatted snippet as follows
+
+.. code-block:: python
+
+    from sklearn.datasets import make_blobs
+
+    n_samples = 100
+    n_components = 3
+    X, y = make_blobs(n_samples=n_samples, centers=n_components)
+
+It is not necessary to create several blocks of code when submitting a bug
+report. Remember other reviewers are going to copy-paste your code and having a
+single cell will make their task easier.
+
+In the section `**Actual results**` of your `Issue template
+<https://github.com/scikit-learn/scikit-learn/blob/main/.github/ISSUE_TEMPLATE/bug_report.yml>`_.
+you are asked to provide the error message including the full traceback of the
+exception. In this case, use the `python-traceback` qualifier. For example::
+
+    ```python-traceback
+    ---------------------------------------------------------------------------
+    TypeError                                 Traceback (most recent call last)
+    <ipython-input-1-a674e682c281> in <module>
+        4 vectorizer = CountVectorizer(input=docs, analyzer='word')
+        5 lda_features = vectorizer.fit_transform(docs)
+    ----> 6 lda_model = LatentDirichletAllocation(
+        7     n_topics=10,
+        8     learning_method='online',
+
+    TypeError: __init__() got an unexpected keyword argument 'n_topics'
+    ```
+
+yields the following when rendered:
+
+.. code-block:: python
+
+    ---------------------------------------------------------------------------
+    TypeError                                 Traceback (most recent call last)
+    <ipython-input-1-a674e682c281> in <module>
+        4 vectorizer = CountVectorizer(input=docs, analyzer='word')
+        5 lda_features = vectorizer.fit_transform(docs)
+    ----> 6 lda_model = LatentDirichletAllocation(
+        7     n_topics=10,
+        8     learning_method='online',
+
+    TypeError: __init__() got an unexpected keyword argument 'n_topics'
+
+The above steps can be implemented in a different order than the progression we
+show in this example. Take into account that having a pipeline that makes sense
+as a model is not important when creating a MCVE.
```

Comment:
I do not understand this sentence. Maybe we can just remove it.

If we want to illustrate that complex preprocessing can be useless to reproduce a problem, I would instead add unnecessary steps to the first code snippets and then add a new section to explain that it's important to try to simplify the model structure to remove any step (pipeline, columntransformers, ...) that are not needed to reproduce the problem.

Reply:
I still don't get it. Maybe the


"""
The simplification steps presented in this guide can be implemented in a different order than the progression we have shown here. The important points are:

- a minimal reproducer should be runnable by a simple copy and paste in a python terminal;
- it should be simplified as much as possible by removing any code steps that are not strictly needed to reproducing the original problem;
- if should ideally only rely on a minimal dataset generated on-the-fly by running the code instead of relying on external data, if possible.
"""

Reply:
But this is not related to markdown formatting. So it would better be moved before the section on markdown formatting.